src/main/java/com/io/image/manager/controller/ImageController.java

```diff
@@ -103,7 +119,7 @@ public ImageController(ImageService imageService, PrometheusMeterRegistry mr, Ap
 
         CacheResult cacheResult;
         try {
-            cacheResult = imageService.fetchAndCacheImage(origin, normalizedFilename, operations, conversionInfo);
+            cacheResult = imageService.fetchAndCacheImage(origin, normalizedFilename, operations, conversionInfo,client);
```

Comment:
```suggestion
            cacheResult = imageService.fetchAndCacheImage(origin, normalizedFilename, operations, conversionInfo, client);
```

Reply:
I also think that all these things shouldn't be in the controller